CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -87,8 +87,45 @@ public class Child {
   /// This is used to e.g. describe the child if all of its tokens are missing in the source file.
   public let nameForDiagnostics: String?
 
-  /// A doc comment describing the child.
-  public let documentation: SwiftSyntax.Trivia
+  /// A summary of a doc comment describing the child. Full docc comment for the
+  /// child is available in ``Child/documentation``, and includes detailed list
+  /// of possible choices for the child if it's a token kind.
+  public let documentationSummary: SwiftSyntax.Trivia
+
+  /// A docc comment describing the child, including the trivia provided when
+  /// initializing the ``Child``, and the list of possible token choices inferred automatically.
+  public var documentation: SwiftSyntax.Trivia {
```

Comment:
The part that I don't quite like about this is that we're essentially templating the doc comment for `Child` in here. But for `Node`, it's in `templates/SyntaxNodeFile.swift`. So this feels like we have a way of storing templates — in `templates`, but this sort of dilutes that distinction. "Don't put logic in your views", as we say in the web apps world.

Reply:
I think perhaps this is a little bit bigger than one occurrence. The pattern is as follows:
- A file in `templates/`, let's say `SyntaxNodesFile.swift` has a result builder that renders the code. Suddenly, you need to generate something that is not easily available as an object property or a function on a `Node` or a `Child`.
- You can't comfortably put that code in the result builder, so you shove that code _somwehere_.
- So far, I've seen extensions of `LayoutNode` (in the other PR of mine) and now I'm doing a function on `Child`. 

That feels slightly unorthodox, but not too bad: 
- templates should be as easy to read as possible. Sure, we're not SwuftUI here, but the closer we can get in terms of readability, the better.
- Node and Child and other definitions should ideally contain the information you need to generate the code, but not the generation code itself — it's like the view layer and data layer to me.
- Seems like Utils / extensions solve that problem partially, but since we're extending Child / Node, it feels like we're injecting view code in the model.


I'm uncertain if my opinion is valid — I might not have enough exposure and experience. But so far, my idea is something like:
- Add `templates/helpers/*.Swift`
- Make `enums` or `structs` that you can't instantiate with static functions that are clean. They take the object in, and render syntax out. They can only depend on their arguments, without side effects.

@ahoppen want me to start a discussion / post on forums on this?

Reply:
I agree that we aren’t super strict on the distinction between the pure data layer and the view layer in CodeGeneration and you probably notice this a lot more than I do because I haven’t been doing much view-based / UI development in a few years, where you think a lot more about that separation.

But then also, the distinction is a little bit muddy IMO. Just taking the documentation as an example. I think we both agree that the documentation should be a property of a node/child but should that documentation already contain the generated *Token* section? I think you can argue in either direction. 

That being said, if you have ideas how to make that distinction clearer, I’m all in favor for them. One thing that’s important to me, though, is that we don’t introduce too many new abstraction layers that make the code less readable. Since CodeGeneration is still fairly reasonable in size and I don’t expect it to grow too much faster in the future, I would prefer an unorthodox decision here and there to an overengineered system of abstraction layers.